csv/komand_csv/actions/to_json/action.py

```diff
@@ -1,32 +1,28 @@
 import komand
-from .schema import ToJsonInput, ToJsonOutput
+from .schema import ToJsonInput, ToJsonOutput, Input, Output, Component
 # Custom imports below
 import base64
 from komand_csv.util import utils
+from komand.exceptions import PluginException
 
 
 class ToJson(komand.Action):
 
     def __init__(self):
         super(self.__class__, self).__init__(
                 name='to_json',
-                description='Convert CSV to JSON',
+                description=Component.DESCRIPTION,
                 input=ToJsonInput(),
                 output=ToJsonOutput())
 
     def run(self, params={}):
-        decoded = base64.b64decode(params['csv'])
+        decoded = base64.b64decode(params[Input.CSV]).decode()
 
-        validation = params.get('validation')
+        validation = params.get(Input.VALIDATION)
         if validation:
             csv_good = utils.csv_syntax_good(decoded)
             if not csv_good:
-                self.logger.error('Malformed CSV')
-                raise Exception
+                raise PluginException(cause='Malformed CSV', assistance='Wrong CSV syntax')
 
         list_of_dicts = utils.csv_to_dict(decoded, self)
-        return {"json": list_of_dicts}
-
-    def test(self):
```

Comment:
Create a test in connection.py.

Reply:
There's a test function in connection.py 👍